pkg/alerts/prometheus.go

```diff
@@ -7,42 +7,77 @@ import (
 	"sort"
 	"time"
 
-	"github.com/prometheus/client_golang/api"
+	log "github.com/sirupsen/logrus"
+
+	papi "github.com/prometheus/client_golang/api"
 	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
 	"github.com/prometheus/common/model"
+	"k8s.io/apimachinery/pkg/api/errors"
 )
 
-// PrometheusActiveAlerts returns a list of names of active (e.g. pending or firing) alerts, filtered
-// by the supplied regexp.
-func PrometheusActiveAlerts(prometheusURL string, filter *regexp.Regexp) ([]string, error) {
-	client, err := api.NewClient(api.Config{Address: prometheusURL})
+// PromClient is a wrapper around the Prometheus Client interface and implements the api
+// This way, the PromClient can be instantiated with the configuration the Client needs, and
+// the ability to use the methods the api has, like Query and so on.
+type PromClient struct {
+	papi papi.Client
+	api  v1.API
+}
+
```

Comment:
yeah, I just dove into this while I was busy with this change.. so I'm not a 'know-it-all' either :) 
but, [https://github.com/prometheus/client_golang/blob/master/api/client.go](prometheus client_golang) using this gives us some defaults regarding the 'Roundtripper' (time-outs and so on).. so I think this is a good way to go. 

Also, I'm not sure if 'token expiration' was in scope before this change either, by using the prometheus client, we can for certainly have some more flexibilty regarding http traffic, like certificates, proxies and so on. 

The client going stale.. Is this a reported issue somewhere? I'm not sure how this could happen all of a sudden :)